src/java/org/apache/cassandra/hints/HintsStore.java

```diff
@@ -76,9 +77,13 @@ private HintsStore(UUID hostId, File hintsDirectory, ImmutableMap<String, Object
         dispatchDequeue = new ConcurrentLinkedDeque<>(descriptors);
         corruptedFiles = new ConcurrentLinkedQueue<>();
         hintsExpirations = new ConcurrentHashMap<>();
+        hintsSizes = new ConcurrentHashMap<>();
 
         //noinspection resource
         lastUsedTimestamp = descriptors.stream().mapToLong(d -> d.timestamp).max().orElse(0L);
+
+        for (HintsDescriptor descriptor : Iterables.concat(dispatchDequeue, corruptedFiles))
```

Comment:
it is not technically necessary to iterate over corrputedFiles because they are empty here